ios/Modules/Components/Sources/Components/States/AppState.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  AppState.swift
+//  iosHackthonApp
+//
+//  Created by Raynelle Francisca on 18/12/2020.
+//  Copyright © 2020 orgName. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+public class AppState: ObservableObject {
+  @Published public var moveToHome: Bool = false
```

Comment:
This feels a bit hacky to me. Sure it works now, but when you have more complex navigations or you just have multiple navigations, you'll end up with a ton of `Bool` flags that you need to take care of every single time.

What about making this a custom class or an enum, something like:
```
public var navDestination: NavDestination? = nil // make it nil as default!
```

Then your NavDestination enum could me something like:
```
enum NavDestination{ 
Home, SectionA, SectionB, SectionC // Add more in the future
}
```

Finally, your view would receive this navDestination like:
```
.onReceive(self.appState.$navDestination) { navDestination in
    // if navDestination is nil, do nothing
    // check here if navDestination is Home, then navigate to home
    // else, check if navDestination is SectionA, then navigate accordingly
    // etc
    // reset navDestination to nil
}
```

I don't know if this would be a good approach for an iOS app, but that definitely would centralise your navigation and it takes care of reusability, scalability and testability.

Reply:
I think this sounds like a nice approach.
It would work in a similar way to how we handle the alerts in the app @raynelle03

Reply:
Umm..tbh not sure how this would work in swiftUI. 😅
I have followed the approach that’s currently out there.
The whole purpose of `moveToHome`  is to basically move to the RootView.

`moveToHome` is an environment object and can be set to true on any view. 

Currently we have 2 flows for login and signup

Login:     LoginView -> HomeScreen->SettingsView
SignUp:  LoginView -> RegisterWithEmailScreen -> EnterTheEmailScreen ->EnterThePasswordScreen ->  HomeScreen-> SettingsView


The `moveToHome` will only be observed on the RootView i.e : `LoginView`. And the only place it would be set , is on the  `SettingsView` When the user taps `Logout button` (So you set it to true only if u want to go back to the root view).
And then accordingly the LoginView navigationLinks `isActive` is set to false. Causing the pop to root view. 

It works for both the flows as we’re only observing it on the RootView (`LoginView`) and setting it on the views we want to pop to the root view.

Even if we have a complex flow(multiple screens), it will still work, as we will only need to set this `moveToHome` to true. and the views should react to this and pop accordingly.

Reply:
@raynelle03 the idea is to keep the same approach that you applied here, but instead of using a `Bool` to decide where you want to go (and then you have to turn that `Bool` to false!), you have a generic Destination object.

My fear is that you might end up with a lot of states like:
`moveToHome`, `moveToDetails`, `moveToSettings`, etc... 
and then, there's the possibility to leave one of those flags to `true` mistakenly and that will produce some weirdness/bugs in the app.

However, if you only have one object that could either be Home, Details or Settings, there's no state to keep/reset and you minimise the probability of bug, minimise the code/states maintenance and you don't have to change your AppState class every time you want a new navigation

Reply:
Navigation in SwiftUI is also something I've not had to handle outside of just using NavigationLinks to move to a screen and back again so I appreciate your concerns.

Reply:
Ahhh okay..think I understand now

Reply:
Okay done...nice one @fmontesino... I din't understand at first😅. But yeah this approach is more scalable!
Added `NavDestination` to appState